test/integration_test.dart

```diff
@@ -1567,6 +1582,46 @@ Future<void> main() async {
     }
   });
 
+  group('Admin page', () {
+    setUp(() async {
+      when(mockAuthProvider.currentUser).thenAnswer((_) => Future.value(User(
+          uid: '0', firstName: 'John', lastName: 'Doe', permissionLevel: 4)));
+      when(mockAuthProvider.currentUserFromCache).thenReturn(User(
+          uid: '0', firstName: 'John', lastName: 'Doe', permissionLevel: 4));
+      when(mockAuthProvider.isAnonymous).thenReturn(false);
+      when(mockAuthProvider.isAuthenticated).thenReturn(true);
+      when(mockAuthProvider.isVerified).thenAnswer((_) => Future.value(true));
+      SharedPreferences.setMockInitialValues({'language': 'auto'});
+      when(mockAdminProvider.fetchUnprocessedRequestsIds())
+          .thenAnswer((_) => Future.value(['string']));
+      when(mockAdminProvider.fetchRequest('')).thenAnswer(
+          (_) => Future.value(Request(requestBody: 'body', userId: '0')));
+    });
+
+    for (final size in screenSizes) {
+      testWidgets('${size.width}x${size.height}', (WidgetTester tester) async {
+        await binding.setSurfaceSize(size);
+
+        await tester.pumpWidget(buildApp());
+        await tester.pumpAndSettle();
+
+        // Open settings page
+        await tester.tap(find.byIcon(Icons.settings_outlined));
+        await tester.pumpAndSettle();
+
+        // Open admin panel page
+        final adminPanelButton = find.byKey(const Key('HandleRequests'));
+        await tester.ensureVisible(adminPanelButton);
+        await tester
+            .pumpAndSettle(); // No idea why this works, (AdrianMargineanu) said so
```

Comment:
Lol, you can remove this comment. It's quite standard to have to call `pumpAndSettle` after most actions in a test.

To explain, basically what `pumpAndSettle` does is it waits for the widgets to "settle" (stop rebuilding). In layman terms, when a human presses a button for example, there's usually a small delay until the action that the button was supposed to do happens. Humans instinctively wait for the thing to finish (a loading spinner to disappear, for example). In testing code on the other hand, if we immediately press a button and then check that the action was performed, the action won't actually have time to happen due to all the animations and all the other stuff that's going on. That's why we need `pumpAndSettle`.

Now, for what `ensureVisible` does. Essentially, depending on screen size, you may need to scroll to press a button in a list. If you test manually you scroll instinctively, but the tester framework doesn't. If the button is not visible on screen, it doesn't know where it is. `ensureVisible` checks the whole widget tree (not just the stuff on the screen) and scrolls to make that widget you wanna press "visible" on the screen.